iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1297,8 +1297,12 @@ public boolean shouldOverwrite(org.apache.hadoop.hive.ql.metadata.Table mTable,
   public Boolean hasAppendsOnly(org.apache.hadoop.hive.ql.metadata.Table hmsTable, SnapshotContext since) {
     TableDesc tableDesc = Utilities.getTableDesc(hmsTable);
     Table table = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
-    boolean foundSince = false;
-    for (Snapshot snapshot : table.snapshots()) {
+    return hasAppendsOnly(table.snapshots(), since);
+  }
+
+  public Boolean hasAppendsOnly(Iterable<Snapshot> snapshots, SnapshotContext since) {
+    boolean foundSince = since == null;
```

Comment:
Maybe it's only me, but the name of this variable and the way it is used for two different things (is since non-null, and did we find the snapshot) is quite confusing. I wouldeither rename it, or split to two variables.

Reply:
if `since` is null all the snapshots has to be checked if it is not only ones which are newer than it.

Reply:
OK, it's very compact and efficient construct, so although it was a bit confusing for me, I'm fine with it.